opentelemetry-sdk/tests/metrics/exponential_histogram/test_exponent_mapping.py

```diff
@@ -0,0 +1,378 @@
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x1000000000000)), -4
+        )  # base == 2 ** -32
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x10000000000000)), -4
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x100000000000000)), -4
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x1000000000000000)), -4
+        )
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x10000000000000000)), -5
+        )  # base == 2 ** -64
+        self.assertEqual(
+            exponent_mapping.map_to_index(1 / float(0x100000000000000000)), -5
+        )
+
+        self.assertEqual(exponent_mapping.map_to_index(float_info.max), 63)
+        self.assertEqual(exponent_mapping.map_to_index(2**1023), 63)
+        self.assertEqual(exponent_mapping.map_to_index(2**1019), 63)
+        self.assertEqual(exponent_mapping.map_to_index(2**1009), 63)
+        self.assertEqual(exponent_mapping.map_to_index(2**1008), 62)
+        self.assertEqual(exponent_mapping.map_to_index(2**1007), 62)
+        self.assertEqual(exponent_mapping.map_to_index(2**1000), 62)
+        self.assertEqual(exponent_mapping.map_to_index(2**993), 62)
+        self.assertEqual(exponent_mapping.map_to_index(2**992), 61)
+        self.assertEqual(exponent_mapping.map_to_index(2**991), 61)
+
+        self.assertEqual(exponent_mapping.map_to_index(2**-1074), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1073), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1072), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1057), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1056), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1041), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1040), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1025), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1024), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1023), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1022), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1009), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1008), -64)
+        self.assertEqual(exponent_mapping.map_to_index(2**-1007), -63)
+        self.assertEqual(exponent_mapping.map_to_index(2**-993), -63)
+        self.assertEqual(exponent_mapping.map_to_index(2**-992), -63)
+        self.assertEqual(exponent_mapping.map_to_index(2**-991), -62)
+        self.assertEqual(exponent_mapping.map_to_index(2**-977), -62)
+        self.assertEqual(exponent_mapping.map_to_index(2**-976), -62)
+        self.assertEqual(exponent_mapping.map_to_index(2**-975), -61)
+
+    def test_exponent_index_mat(self):
```

Comment:
what is mat?

Reply:
No idea, fixed. :stuck_out_tongue_closed_eyes: